### ISSUE TITLE
Incorrect FEE temperature (from file)

### ISSUE THREAD

**Issue by SaraRegibo** [IvS-KULeuven/PlatoSim3#413]
- Keep the CCD at nominal operating temperature.
- Let the operating temperature increase above the nominal operating temperature.  E.g. from 210.15K to 220.15K over the course of 100 exposures.
- In the methods `getGainLeftAdc()` and `getGainRightAdc()` add log statements to show the temperature of the FEE and the deviation from the  nominal operating temperature, when the gain is determined, e.g.
Log.info("FEE temperature (left): " + to_string(getTemperature(time)) + " " + to_string(getTemperature(time) - nominalOperatingTemperature)); ```

For the first exposure I see this in my log file:

2020-01-14 15:46:08 INFO    FEE temperature (left): 210.150000 0.000000
2020-01-14 15:46:08 INFO    FEE temperature (right): 210.150000 0.000000

and for the last exposure (100'th):

2020-01-14 15:46:08 INFO    FEE temperature (left): 220.050000 0.000000
2020-01-14 15:46:08 INFO    FEE temperature (right): 210.150000 0.000000

The problems:
- Despite the fact that the temperature of the "left" FEE changes correctly, the reported difference with the nominal operating temperature is incorrect (i.e. it stays zero, instead of being 10K).
- The temperature does not change for the "right" FEE.

See also the `Gain` notebook in the `validation-verification` branch in my personal repository.

**Comment by SaraRegibo** [comment]
Blocks successful testing of:
- PS-GAIN-2
- PS-OFF-1

**Comment by SaraRegibo** [comment]
I have a simulation of 100 exposures, over which the temperature of the FEE increases with 10K:

```
deltaTemperature = 10
time = np.array([0, numExposures * 25])

temperature = np.array([sim["FEE/NominalOperatingTemperature"], sim["FEE/NominalOperatingTemperature"] + deltaTemperature])
```
This means that the FEE gain and electronic offset should change accordingly.  I keep all other effects disabled (so we apply the gain and electronic offset but no noise).

First, the FEE gain is applied.  This means that the temperature of the FEE is asked for and the temperature-dependent FEE gain is calculated and applied.  However, when doing this, the internal time of  the `TemperatureFromFile` class is advanced 25s (whereas the `timeIndex` stays zero).  So when you then ask for the temperature of the FEE when calculating the temperature-dependent electronic offset, you end up in a situation where

```
time == internalTime
```
Consequently the 1st entry in the temperature file is returned every time you ask for the FEE temperature from within `addElectronicOffset` (since the internal time of `TemperatureFromFile` has already been advanced by `applyGain` whilst `timeIndex` stays at zero).

**Comment by SaraRegibo** [comment]
- Solved in #466 
- Validated in #467
